src/Connections/GenericHook.ts

```diff
@@ -258,18 +264,41 @@ export class GenericHookConnection extends BaseConnection implements IConnection
         }
     }
 
-    public async onStateUpdate(stateEv: MatrixEvent<unknown>) {
-        const validatedConfig = GenericHookConnection.validateState(stateEv.content as Record<string, unknown>, this.config.allowJsTransformationFunctions || false);
-        if (validatedConfig.transformationFunction) {
-            try {
-                this.transformationFunction = new Script(validatedConfig.transformationFunction);
-            } catch (ex) {
-                await this.messageClient.sendMatrixText(this.roomId, 'Could not compile transformation function:' + ex, "m.text", this.intent.userId);
-            }
+    public onStateUpdate(stateEv: MatrixEvent<unknown>) {
+        return this.updateConfig(stateEv.content as Record<string, unknown>);
+    }
+
+    private async updateConfig(config: Record<string, unknown>) {
+        this.state = {
+            ...GenericHookConnection.validateState(config),
+            hookId: this.state.hookId,
+        };
+        return this.validateTransformationFunction();
+    }
+
+    private validateTransformationFunction(): Promise<void> {
+        let result = Promise.resolve();
+        if (!GenericHookConnection.quickModule || !this.state.transformationFunction) {
+            return result;
+        }
+        this.hasValidTransformation = false;
+        const ctx = GenericHookConnection.quickModule.newContext();
+        const codeEvalResult = ctx.evalCode(`function f(data) {${this.state.transformationFunction}}`);
+        if (codeEvalResult.error) {
```

Comment:
I believe this a breaking change also, we never validated the function until the webhook ran before, and simply let the user know when there was an error.

I'm not saying this isn't a good way to go in the future, but I'd rather not break behaviours in this PR.

Reply:
It got validated on a state event update, and using the provisioning API would both call the constructor & send a state event which Hookshot would see as an update.  So in practice, it did get validated before being run, except in some tests that only call the constructor without sending a state event.

Reply:
Reverted, though it retains the inconsistency of how the transformation function is validated & set between state updates & the constructor.  We can revisit that in a later change, though.